letter-shop.md

```diff
@@ -0,0 +1,81 @@
+# Letter Shop Tutorial
+
+## What you need before you start:
+
+* a laptop with an internet connection and NodeJS installed
+* basic knowledge of the command line terminal
+* basic knowledge of JavaScript, including Promises and Buffers
+* (optional) the concept of one-way hash functions
+
+## What you'll learn:
+
+* the hashlock concept
+* the Ledger Plugin Interface (LPI)
+* how to build a paid web app using Interledger!
```

Comment:
I like that you don't have to copy and paste the code anymore but it almost feels too quick now, like it's missing the actual code walkthrough. What do you think about the tutorial style of something like [React](https://facebook.github.io/react/tutorial/tutorial.html) where they show small snippets next to explanation and then also provide the full code of the example? (They also link to codepen, which is cool cause there's no need to download it to play around with it, but then again it is a frontend framework)

Reply:
> What do you think about the tutorial style of something like React where they show small snippets next to explanation and then also provide the full code of the example?

That's my preferred style of tutorial. Snippets inside code and then full code at the end.

The Kotlin Tutorials have a very cool way of showing code snippets that you can expand and execute: https://kotlinlang.org/docs/reference/basic-syntax.html

Not sure what library they use.